.github/workflows/native-build.yml

```diff
@@ -100,5 +100,9 @@ jobs:
       - name: Running and printing the benchmark examples.
         run: ${{github.workspace}}/build/benchmark/BenchmarkExamples -p
 
-      - name: E2E
+      - name: E2E with index in one go
+        if: (matrix.compiler == 'clang')
         run: ${{github.workspace}}/e2e/e2e.sh
+      - name: E2E with separate text index build
+        if: (matrix.compiler != 'clang')
+        run: ${{github.workspace}}/e2e/e2e.sh -t
```

Comment:
I assume this is just an arbitrary split to test the two variants without multiplying the time the E2E tests take with a factor of 2. If yes, please add a comment, if no please also add a comment explaining why it depends on the compiler.